### ISSUE TITLE
Training with known poses vs COLMAP

### ISSUE THREAD

**Issue by cta5425** [nerfstudio-project/nerfstudio#3473]
Hi, 
I used a simulator to generate training data with known poses. The images are actually so realistic, that its hard to realize that its synthetic. 

I have tried to train a Nerfacto with known poses using the transforms.json. But the model results in a scattered scene, I can see the form but its not accurate. I know (From previous issues) that Nerfacto does not work well with synthetic data, but it shows very good results when I process the data using ns-process-data and ignore my known poses... 

Why are the models created by Nerfacto or Splatfacto turn to be good, if used after running ns-process-data and not using the known poses.

I am looking forward to your answer. 

Best regards,

**Comment by maturk** [comment]
Hi @cta5425 , it would help very much if you added pictures of what your poses look like in e.g. viser/viewer compared to the colmap poses. There are many small issues that can happen if you are manually making your own transforms.json file, the most common is just using the wrong camera conventions when defining the c2w matrices.

There should be no reason that ground truth poses would not work with nerfstudio. In fact, many datasets like Replica are also synthetic datasets that contain ground truth poses and they work just fine with nerfacto / splatfacto.

**Comment by cta5425** [comment]
Hi @maturk , many thanks for your prompt response! I faced many issues when making my own transforms.json regarding c2w matrices as you mentioned, but I thought I managed to solve them. Below are 2 picture from a trained Nerfacto, first time with my own created transforms.json, second time poses are estimated using colmap (ns-process-data). 
1- 
![Screenshot from 2024-10-11 15-02-27](https://github.com/user-attachments/assets/287b9b3b-b4c7-471b-81dd-8f70f32a2fd5)
2- 
![Screenshot from 2024-10-11 15-01-58](https://github.com/user-attachments/assets/995657c8-fb59-4ab2-bef4-4a5fb97a2357)

I checked the convention carefully to convert to be able to get the c2w matrices, yet I am somehow unsure when I see the results, I could be missing something.

**Comment by maturk** [comment]
@cta5425 One questions, have you checked that both splatfacto and nerfacto fail with this scene?

One note that came to mind is that nerfacto expects poses to be within a [-1,1] unit cube. If the poses are not scaled to fit in the unit cube, the reconstruction might suffer. (flag for that is [here](https://github.com/nerfstudio-project/nerfstudio/blob/67be06009eae045f25b1ce5a67c012cbaf57e2ab/nerfstudio/data/dataparsers/nerfstudio_dataparser.py#L60)).

The poses of your cameras look identical to the colmap ones, therefore, I would investigate the intrinsics K of your custom data. Are they correct? Note that colmap estimates focals fx and fy in pixel units, I am not sure what focal estimates you have for your custom transforms.json.

**Comment by isso313** [comment]
@maturk Many thanks for your response!!! Your tipp was actually correct, I investigated the instrinsics K of my custom data and compared the ones I calculated with the extracted ones from COLMAP. 

It turned out I assumed my camera model which is a "PINHOLE" has two different focals fx and fy. In fact the focals fx and fy were equal, since no distortion is modeled, as COLMAP estimated. 

This result is with my own written transforms.json:
<img width="1904" alt="test-known-poses" src="https://github.com/user-attachments/assets/91f78e51-7497-4654-9104-f7b05f509ff0">

Many thanks again, and congratulations on your new work dn-splatter, I am definitely going to try it out!

**Comment by cta5425** [comment]
@maturk Many thanks for your response!!! Your tipp was actually correct, I investigated the instrinsics K of my custom data and compared the ones I calculated with the extracted ones from COLMAP. 

It turned out I assumed my camera model which is a "PINHOLE" has two different focals fx and fy. In fact the focals fx and fy were equal, since no distortion is modeled, as COLMAP estimated. 

This result is with my own written transforms.json:
<img width="1904" alt="test-known-poses" src="https://github.com/user-attachments/assets/91f78e51-7497-4654-9104-f7b05f509ff0">

Many thanks again, and congratulations on your new work dn-splatter, I am definitely going to try it out!
